denote.el

```diff
@@ -3933,6 +3937,143 @@ To be assigned to `markdown-follow-link-functions'."
 (eval-after-load 'markdown-mode
   '(add-hook 'markdown-follow-link-functions #'denote-link-markdown-follow))
 
+;;;;; Link fontification
+(defcustom denote-link-display-style 'button
+  "The style for displaying denote links.
+
+By default (`button'), the link is displayed as a button. This
+requires calling `denote-link-buttonize-buffer' or enabling the
+minor-mode `denote-fontify-links-mode' appropriately. See the
+documentation of that function for more details.
+
+When the value is `fontify', the link is fontified using
+font-lock. The appropriate keyword must be added to `font-lock'.
+Enabling the minor-mode `denote-fontify-links-mode' does this.
+
+The value `fontify-and-fold', is the same as `fontify' but also
+uses `org-fold' (if available) to fold the links.
+"
+  :type '(choice
+          (const :tag "As button" button)
+          (const :tag "Fontified" fontify)
+          (const :tag "Fontified and folded" fontify-and-fold))
+  :package-version '(denote . "0.6.0")
+  :group 'denote)
+
+(defvar denote-link-keymap
+  (let ((map (make-sparse-keymap)))
+    (define-key map [mouse-2] #'denote-link-open-at-mouse)
+    (define-key map [mouse-3] #'denote-link-open-at-mouse)
+    (define-key map [follow-link] #'mouse-face)
```

Comment:
Is the `mouse-face` a function? I cannot find it on my end (Emacs 30).

Reply:
You're right. It's not defined on mine either. This keymap is mirroring one that I got from `org` (you can check `org-mouse-map` on your setup).

Reply:
See [here](https://www.gnu.org/software/emacs/manual/html_node/elisp/Clickable-Text.html) for an explanation of `mouse-face`.